great_expectations/agent/agent_server_session.py

```diff
@@ -0,0 +1,27 @@
+import pydantic
+from requests import Session
+
+from great_expectations.core.http import create_session
+
+
+class GxAgentServerSessionConfig(pydantic.BaseSettings):
+    gx_cloud_base_url: str = "https://api.greatexpectations.io"
```

Comment:
```suggestion
    gx_cloud_base_url: pydantic.HttpUrl = "https://api.greatexpectations.io" # type: ignore[assignment] # str will be coerced
```
https://docs.pydantic.dev/1.10/usage/types/#urls

I would suggest a `HttpUrl` type here. Raise issues early related to bad URL format.

Reply:
this doesn't work because `HttpUrl` doesn't allow `localhost` + a port, which we need for local dev work.
```python
from pydantic import BaseModel, HttpUrl
class A(BaseModel):
    url: HttpUrl

a = A("http://localhost:5000")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pydantic/main.py", line 332, in pydantic.main.BaseModel.__init__
TypeError: __init__() takes exactly 1 positional argument (2 given)
>>> a = A(url="http://localhost:5000")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for A
url
  URL host invalid, top level domain required (type=value_error.url.host)
```

Reply:
You can use `AnyUrl`

Reply:
ah great, thanks!